src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -205,6 +205,22 @@ public static IResourceBuilder<T> WithPgAdmin<T>(this IResourceBuilder<T> builde
         }
     }
 
+    /// <summary>
+    /// Configures the host port that the PostgresServer resource is exposed on instead of using randomly assigned port.
+    /// </summary>
+    /// <param name="builder">The resource builder for PostgresServer.</param>
+    /// <param name="port">The port to bind on the host. If <see langword="null"/> is used random port will be assigned.</param>
+    /// <returns>The resource builder for PostgresServer.</returns>
+    public static IResourceBuilder<PostgresServerResource> WithHostPort(this IResourceBuilder<PostgresServerResource> builder, int? port)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder.WithEndpoint("http", endpoint =>
```

Comment:
I don't think `http` is the right endpoint name in this case. I think you want 'tcp'.

Reply:
Yes sorry, noticed right after and pushed the changes.